### PR TITLE
chore/simplified urls

### PR DESCRIPTION
**Pull request by gecBurton**
## Context

As a user i want to have working links

## Changes proposed in this pull request

I have also taken the opportunity to clean up the url code

## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [ ] I have tested any code added or changed
- [x] I have run integration tests https://github.com/i-dot-ai/redbox/actions/runs/11592463598

### CODE REVIEW

File: django_app/redbox_app/redbox_core/models.py [hunk 1]
```diff
@@ -818,9 +792,9 @@ def save(self, force_insert=False, force_update=False, using=None, update_fields
         super().save(force_insert, force_update, using, update_fields)
 
     @property
-    def uri(self) -> str:
-        """returns either the url of an external citation or the file uri of a user-uploaded document"""
-        return self.url or f"file://{self.file.file_name}"
+    def uri(self) -> URL:
+        """returns the url of either the external citation or the user-uploaded document"""
+        return URL(self.url or self.file.url)
```
Comment (gecBurton) [inline django_app/redbox_app/redbox_core/models.py:797]: this is the key change

File: django_app/redbox_app/redbox_core/models.py [hunk 2]
```diff
@@ -614,32 +612,8 @@ def file_type(self) -> str:
         return name.split(".")[-1]
 
     @property
-    def url(self) -> URL | None:
-        #  In dev environment, get pre-signed url from minio
-        if settings.ENVIRONMENT.uses_minio:
```
Comment (KevinEtchells) [inline django_app/redbox_app/redbox_core/models.py:619]: Presuming you no longer need this in dev?

**Review by KevinEtchells — APPROVED** [review]
Just one question I've added inline. But LGTM!

File: django_app/redbox_app/redbox_core/models.py (hunk 2 above)
Comment (gecBurton) [reply django_app/redbox_app/redbox_core/models.py:619]: we dont seem to - integration tests and unit tests look fine

**Pull request merged**